zmq/backend/cython/socket.pyx

```diff
@@ -679,7 +758,54 @@ cdef class Socket:
                 msg = Frame(data, track=track)
             return _send_frame(self.handle, msg, flags)
 
-    cpdef object recv(self, int flags=0, copy=True, track=False):
+    cpdef object send_multipart(self, msg_parts, int flags=0, bint copy=True, bint track=False):
+        """send a sequence of buffers as a multipart message
+
+        The zmq.SNDMORE flag is added to all msg parts before the last.
+
+        Parameters
+        ----------
+        msg_parts : iterable
+            A sequence of objects to send as a multipart message. Each element
+            can be any sendable object (Frame, bytes, buffer-providers)
+        flags : int, optional
+            SNDMORE is handled automatically for frames before the last.
+        copy : bool, optional
+            Should the frame(s) be sent in a copying or non-copying manner.
+        track : bool, optional
+            Should the frame(s) be tracked for notification that ZMQ has
+            finished with it (ignored if copy=True).
+
+        Returns
+        -------
+        None : if copy or not track
+        MessageTracker : if track and not copy
+            a MessageTracker object, whose `pending` property will
+            be True until the last send is completed.
+        """
+        if copy:
+            return _send_multipart_copy(self.handle, msg_parts, flags)
+        else:
+            for data in msg_parts[:-1]:
+                if isinstance(data, Frame):
+                    if track and not data.tracker:
+                        raise ValueError('Not a tracked message')
+                    msg = data
+                else:
+                    msg = Frame(data, track=track)
+                _send_frame(self.handle, msg, flags|ZMQ_SNDMORE)
+
+            data = msg_parts[-1]
```

Comment:
The original implementation in sugar.py doesn't. I guess it's because it really has to fail with empty messages, you can't send zero-part multipart messages in zmq.